pennylane_lightning/src/simulators/lightning_qubit/utils/SparseLinAlg.hpp

```diff
@@ -182,52 +181,27 @@ namespace Pennylane::LightningQubit::Util {
  * @param entries_ptr   pointer to the column indices of the non-zero elements.
  * @param values_ptr    non-zero elements.
  * @param numNNZ        number of non-zero elements.
- * @param result        result of the matrix vector multiplication.
+ * @return result       result of the matrix vector multiplication.
  */
 template <class fp_precision, class index_type>
-void apply_Sparse_Matrix_Kokkos(
-    [[maybe_unused]] const std::complex<fp_precision> *vector_ptr,
-    [[maybe_unused]] const index_type vector_size,
-    [[maybe_unused]] const index_type *row_map_ptr,
-    [[maybe_unused]] const index_type row_map_size,
-    [[maybe_unused]] const index_type *entries_ptr,
-    [[maybe_unused]] const std::complex<fp_precision> *values_ptr,
-    [[maybe_unused]] const index_type numNNZ,
-    [[maybe_unused]] std::vector<std::complex<fp_precision>> &result) {
-    PL_ABORT("Executing the product of a Sparse matrix and a vector needs "
-             "Kokkos and Kokkos Kernels installation.");
+std::vector<std::complex<fp_precision>>
+apply_Sparse_Matrix(const std::complex<fp_precision> *vector_ptr,
+                    const index_type vector_size, const index_type *row_map_ptr,
+                    [[maybe_unused]] const index_type row_map_size,
+                    const index_type *entries_ptr,
+                    const std::complex<fp_precision> *values_ptr,
```

Comment:
We could likely replace many of these with a [`std::span`](https://en.cppreference.com/w/cpp/container/span) to keep the pointer and size in the same location. If it is too painful right now, we can add this in a follow up.

Reply:
The source code would benefit from an `std::span` job across the board, but I think this may be done in another PR.